skimage/io/_io.py

```diff
@@ -1,31 +1,34 @@
-from io import BytesIO
-
 import numpy as np
 import six
 
 from ..io.manage_plugins import call_plugin
 from ..color import rgb2grey
 from .util import file_or_url_context
 from ..exposure import is_low_contrast
-from .._shared.utils import all_warnings, warn
+from .._shared.utils import warn
 
 
 __all__ = ['imread', 'imsave', 'imshow', 'show',
            'imread_collection', 'imshow_collection']
 
 
-def imread(fname, as_grey=False, plugin=None, flatten=None,
+def imread(fname, as_gray=False, as_grey=None, plugin=None, flatten=None,
```

Comment:
Hmm, this will break for users who called `imread` with positional arguments.

I suggest that we use `kwargs` to catch `as_grey` *indefinitely*. Or was there a decision made that we don't want to support both versions? My argument is that since we have kwargs here anyway, it is not a big cost: autocomplete will always provide the "canonical" `as_gray`, but if a user types in `as_grey` it will just work. You would just need the line:

```python
as_gray = kwargs.pop('as_grey', as_gray)
```